### PR TITLE
One log

### LINKED ISSUE #16585: Does Elasticsearch need to include wrappers for java.util.logging and slf4j?
Right now Elasticsearch has three logger implementations - one for log4j, one for java.util.logging, and one for slf4j. What does that buy us? The elasticsearch zip, tar, deb, and rpm distributions bundle log4j and logging.yaml doesn't work without it. When Elaticsearch is used as a client library master no longer tries to configure the logger because that is crazy behavior for a library. Normal Java libraries just pick a logging dependencies and use it. They rely on the user to include whatever bridges are required. Can we get away with being normal?

Proposal: Remove support for java.util.logging and slf4j logging support. We fold the log4j support into its parent class/package. We put a breaking change note for users of the TransportClient that they'll need to include the appropriate shim to point log4j12 at whatever logging library they use. They probably already have it on their classpath. We get to nuke three whole packages, lots of weird try/catch statements, and 2 optional dependencies.

If anyone asks why we still use log4j12 when log4j2 and logback are all around we tell them that it works and we have bigger fish to fry....

### PR DESCRIPTION
**Pull request by nik9000**
Removes all Elasticsearch logger wrappers exception log4j. There was some discussion in #16585 around removing all but j.u.l so that we could remove the log4j dependencies, and we may still do that, but I think that we should do this first. My reasoning is that after this it should be reasonably easy to switch the backing logger. The hard work of cutting over to j.u.l would be supporting the logging configuration we support now. Or making a breaking change to support whatever we can get out of j.u.l.

Closes #16585

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java [hunk 1]
```diff
@@ -19,62 +19,29 @@
...
-    }
-
+    public static final Setting<LogLevel> LOG_DEFAULT_LEVEL_SETTING =
+            new Setting<>("logger.level", LogLevel.INFO.name(), LogLevel::parse, false, Setting.Scope.CLUSTER);
+    public static final Setting<LogLevel> LOG_LEVEL_SETTING =
+            Setting.dynamicKeySetting("logger.", LogLevel.INFO.name(), LogLevel::parse, true, Setting.Scope.CLUSTER);
 
     public static ESLogger getLogger(String prefix, String name) {
-        return defaultFactory.newInstance(prefix == null ? null : prefix.intern(), name.intern());
+        prefix = prefix == null ? null : prefix.intern();
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java]: Can we drop this interning?

File: core/src/main/java/org/elasticsearch/common/logging/LogConfigurator.java [hunk 2]
```diff
@@ -39,13 +39,14 @@
...
 import java.util.Properties;
+import java.util.Set;
 
 import static java.util.Collections.unmodifiableMap;
 import static org.elasticsearch.common.Strings.cleanPath;
 import static org.elasticsearch.common.settings.Settings.settingsBuilder;
 
 /**
- *
+ * Configures log4j with a specila set of replacements.
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/common/logging/LogConfigurator.java]: typo: specila -> special

**Comment by rjernst** [comment]
LGTM, we should continue down this path of simplifying ES logging....

File: core/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java (hunk 1 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java]: Probably! I don't know why it was there and it was undocumented. I'm inclined to leave it in this PR and open an issue for it so we don't forget it entirely.

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java]: #16831

**Comment by nik9000** [comment]
I'm going to squash/rebase and merge this if the rebase is clean. If we want j.u.l support one day we can resurrect it from that commit this becomes.

**Comment by nik9000** [comment]
Rebase was clean (one merge error in the imports...). Rerunning tests now.

**Comment by clintongormley** [comment]
@nik9000 i'm assuming this is a breaking change? If so, please add a note to breaking changes (and label this PR)

**Pull request merged**